### PR TITLE
256 wrap native asset

### PR DESCRIPTION
**Pull request by adub671**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx [hunk 1]
```diff
@@ -240,6 +254,27 @@ const AmountSelect = ({
...
 
+  const onWrap = async () => {
+    try {
+      const tx: TransactionRequest = {
+        to:
+          currentChain.id === 1337 ? ganacheWTokenAddress : midasSdk.chainSpecificAddresses.W_TOKEN,
+        value: amount,
+        from: address,
+        nonce: await midasSdk.signer.getTransactionCount(),
+        gasLimit: utils.hexlify(100000),
```
Comment (carlomazzaferro) [inline packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx:265]: Not sure if we should specify the gas limit here -- @peetzweg any ideas if we should keep this?

File: packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx [hunk 2]
```diff
@@ -720,7 +794,12 @@ const TokenNameAndMaxButton = ({
     setIsLoading(true);
 
     try {
-      const maxBN = (await fetchMaxAmount(mode, midasSdk, address, asset)) as BigNumber;
+      let maxBN;
+      if (optionToWrap) {
+        maxBN = await midasSdk.signer.getBalance();
```
Comment (carlomazzaferro) [inline packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx:799]: Nice one!

**Comment by peetzweg** [comment]
Looks pretty good! 👏 
Of course some thorough testing would be appreciated, will give you the private key for the testing wallet on discord.

Unfortunately the `AmountSelect` component is quite a mess and needs refactor itself. So the way you did it here is totally fine with me.

We should probably add some test with dappateer tests one this is merged:
https://github.com/Midas-Protocol/monorepo/pull/628

**Review by peetzweg — APPROVED** [review]

**Comment by smartdevcom** [comment]
Sure, I will add it once merged

File: packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx (hunk 1 above)
Comment (peetzweg) [reply packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx:265]: Haven't read the weth docs yet. But could imagine that there is a suggested gaslimit?

Comment (carlomazzaferro) [reply packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx:265]: I suppose the best practice is to estimate the gasLimit: https://stackoverflow.com/a/72068272/2896045 

I am not sure we do it elsewhere, but I believe that'd be good to follow.

File: packages/sdk/src/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export { default as MidasSdk } from "./MidasSdk";
 export { default as ERC20Abi } from "./MidasSdk/abi/ERC20.json";
+export { default as WETHAbi  } from "./MidasSdk/abi/WETH.json"
```
Comment (carlomazzaferro) [inline packages/sdk/src/index.ts:3]: extra space after WETHAbi that's causing linting errors!

File: packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx [hunk 4]
```diff
@@ -240,6 +252,41 @@ const AmountSelect = ({
...
+      // const tx: TransactionRequest = {
+      //   to:
+      //     currentChain.id === 1337 ? ganacheWTokenAddress : midasSdk.chainSpecificAddresses.W_TOKEN,
+      //   value: amount,
+      //   from: address,
+      //   nonce: await midasSdk.signer.getTransactionCount(),
+      //   gasLimit: utils.hexlify(100000),
+      //   gasPrice: await midasSdk.signer.getGasPrice(),
+      // };
+    const WToken = getContract(midasSdk.chainSpecificAddresses.W_TOKEN, WETHAbi, midasSdk.signer);
```
Comment (carlomazzaferro) [inline packages/ui/components/pages/Fuse/Modals/PoolModal/AmountSelect.tsx:266]: nitpick: for variable names, feel free to use `wToken` (lowercase first letter)

**Comment by carlomazzaferro** [comment]
Other: remove the logging stuff and the commented out code, and test this on BSC fork. Then, we merge!

**Pull request merged**